plugins/versionpress/tests/Unit/QueryLanguageUtilsTest.php

```diff
@@ -178,12 +179,12 @@ public function rulesAndGitLogQueryProvider()
                 '-i --all-match --grep="^VP-Action: \(entity\|.*\)/.*/\(.*vp.*\|vpid\)$"'
             ],
             [['text' => ['text1', 'Test text', '*']], '-i --all-match --grep="text1" --grep="Test text" --grep=".*"'],
-            [['x-vp-another-key' => ['Test value']], '-i --all-match --grep="^x-vp-another-key: \(Test value\)$"'],
-            [['vp-another-key' => ['Test value']], '-i --all-match --grep="^\(x-\)\?vp-another-key: \(Test value\)$"'],
-            [['another-key' => ['Test value']], '-i --all-match --grep="^\(x-vp-\|vp-\)another-key: \(Test value\)$"'],
+            [['X-vp-another-key' => ['Test value']], '-i --all-match --grep="^X-VP-another-key: \(Test value\)$"'],
```

Comment:
I think that no case juggling is necessary here. If the input is `X-vp-another-key`, Git search should be `... --grep="^X-vp-another-key` as well.

Similarly for the lines below.

Reply:
Yes, but we need it for `QueryLanguageUtils::createGitLogQueryFromRule` and also this test (and I believe that it was the original reason to make tests case-insensitive before).
Example:  `vp-another-key` was automatically transformed into `X-vp-another-key` and test would fail because of expecting `x-vp-another-key`.
https://github.com/versionpress/versionpress/blob/c7695ab4cb573fc81376cb984b127657d3cec523/plugins/versionpress/src/Utils/QueryLanguageUtils.php#L231
Result will be now capitalized everytime: `X-VP-another-key` (see changes in `QueryLanguageUtils::createGitLogQueryFromRule`). But I am not sure if this is the best solution, see my previous comment: https://github.com/versionpress/versionpress/pull/1444#pullrequestreview-236416513.